### DISCUSSION TITLE
how to build source code in windows?

### DISCUSSION THREAD

**Discussion by ilovenili** [snl-dakota/dakota#81] [Q&A]
HI,ALL:
I only want to build source code in vscode ,not visual studio c++.
I have try,but failure, please help me !

**Comment by jadamstephens** [comment]
Hi,

Our only experience with building Dakota on Windows is with Visual Studio and Intel (for Fortran). Regrettably, I cannot offer any advice about using other tools.

Adam

**Reply by ilovenili** [reply]
great,,can you tell me the detail step?

**Comment by briadam** [comment]
There aren't comprehensive, up-to-date directions for Windows in the user manual, but the idea roughly follows that described here: https://snl-dakota.github.io/docs/6.19.0/users/compiling/compiling.html, e.g., 

- https://snl-dakota.github.io/docs/6.19.0/users/compiling/devenv.html#windows
- https://snl-dakota.github.io/docs/6.19.0/users/compiling/examples.html#windows-10

Note that compiler and tool versions will have to be updated and properly paired for compatibility in your environment, so what's there likely won't work verbatim.

**Comment by briadam** [comment]
I believe the currently published Dakota Windows binaries use versions similar to these:

- MS VS 2019 16.7.2 
- Intel Visual Fortran 2020 update 4
- Specific version shouldn't much matter: Cmake, Python3 including Sphinx for docs, Perl, JDK11 (Adoptium)
- Boost pre-built binaries, e.g., https://sourceforge.net/projects/boost/files/boost-binaries/1.71.0/boost_1_71_0-msvc-14.2-64.exe/download
- HDF5, custom built from source code using CMake and MSVS
- Reference BLAS/LAPACK built from source using CMake and MSVS/Intel Fortran: https://github.com/Reference-LAPACK/lapack/archive/refs/tags/v3.10.1.tar.gz
